packages/logger/index.js

```diff
@@ -0,0 +1,123 @@
+const logger = require('../dd-trace/src/log')
+// path to require tracer logger
+
+const https = require('https')
+const coalesce = require('koalas')
+
+class V2LogWriter {
+  // Note: these attribute names match the corresponding entry in the JSON payload.
+  constructor ({ ddsource, hostname, service, site, apiKey, interval, timeout }) {
+    this.ddsource = ddsource
+    this.hostname = hostname
+    this.service = service
+    this.interval = interval
+    this.timeout = timeout
+    this.buffer = []
+    this.buffer_limit = 1000
+    this.apiKey = apiKey
+    this.endpoint = '/api/v2/logs'
+    this.site = coalesce(site, 'datadoghq.com')
+    this.intake = `http-intake.logs.${this.site}`
+    this.headers = {
+      'DD-API-KEY': this.apiKey,
+      'Content-Type': 'application/json'
+    }
+    this.timer = setInterval(() => {
+      this.flush()
+    }, this.interval)
+
+    logger.debug(`"started log writer to ${this.url}"`)
+  }
+
+  start () {
+    this.timer()
+  }
+
+  tagString (tags) {
+    let tagStr = ''
+    for (const key in tags) {
+      tagStr += key + ':' + tags[key] + ','
+    }
+    return tagStr
+  }
+
+  log (log, tags, span) {
+    const logTags = this.tagString(tags)
+    const toLog = {
+      'timestamp': Date.now(),
+      'message': log.message,
+      'hostname': coalesce(log.hostname, this.hostname),
+      'ddsource': coalesce(log.ddsource, this.ddsource),
+      'service': coalesce(log.service, this.service),
+      'status': log.level,
+      'ddtags': logTags,
```

Comment:
In the case that no tags are provided the resulting JSON payload would probably have an entry like this:

```json
"ddtags": "",
```

However, it would be nice to simply not include the line at all since it's probably just wasted bytes sent over the network.

An easy way to omit the value would be to rewrite the line like this:

```js
      'ddtags': logTags || undefined,
```

That would then replace an empty string with `undefined`, and the `JSON.stringify()` call would then omit the key from the resulting JSON.

That said, you might want to test sending a log to the backend and view it on the dashboard just to make sure it still works.